Rename PythonUDF{Call,Resp}

pietern opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27531 Run clang-format for torch/distributed/rpc
* **#27530 Rename PythonUDF{Call,Resp}**
* #27529 Scope pybind11 functions to torch.distributed.{autograd,rpc}

Per discussion in #27286, the `UDF` part is superfluous.

This makes the naming consistent with the `MessageType` enum.

Differential Revision: [D17808211](https://our.internmc.facebook.com/intern/diff/D17808211)

pietern committed:
Rename PythonUDF{Call,Resp}

Per discussion in #27286, the `UDF` part is superfluous.

This makes the naming consistent with the `MessageType` enum.

pietern committed:
Update on "Rename PythonUDF{Call,Resp}"


Per discussion in #27286, the `UDF` part is superfluous.

This makes the naming consistent with the `MessageType` enum.

mrshenli reviewed (approved):
Stamp the changes on the names, but tests are broken since the PR before this one. Please fix before landing.

pietern committed:
Update on "Rename PythonUDF{Call,Resp}"


Per discussion in #27286, the `UDF` part is superfluous.

This makes the naming consistent with the `MessageType` enum.

Differential Revision: [D17808211](https://our.internmc.facebook.com/intern/diff/D17808211)

pietern committed:
Update on "Rename PythonUDF{Call,Resp}"


Per discussion in #27286, the `UDF` part is superfluous.

This makes the naming consistent with the `MessageType` enum.

Differential Revision: [D17808211](https://our.internmc.facebook.com/intern/diff/D17808211)

pietern committed:
Update on "Rename PythonUDF{Call,Resp}"


Per discussion in #27286, the `UDF` part is superfluous.

This makes the naming consistent with the `MessageType` enum.

Differential Revision: [D17808211](https://our.internmc.facebook.com/intern/diff/D17808211)

pietern committed:
Update on "Rename PythonUDF{Call,Resp}"


Per discussion in #27286, the `UDF` part is superfluous.

This makes the naming consistent with the `MessageType` enum.

Differential Revision: [D17808211](https://our.internmc.facebook.com/intern/diff/D17808211)

pietern commented:
The commit that's landing has a fix for the build failures in the most recent commit in this PR.
